### DISCUSSION TITLE
Språkvalg

### DISCUSSION THREAD

**Discussion by mrosvik** [digdir/designsystemet#1682] [Patterns for public services]
Relevant:
- https://github.com/digdir/innsiktsbibliotek/issues/16

Det er varierende plassering av språkvalg og ikonbruk. Språkikon kan være flagg (Skatt), en globe (Nav) eller mangle på forsiden (HelseNorge).

**Comment by solhell** [comment]
Et par innledenden refleksjoner her:
- Inngangen til globale språkvalg bør ligge i header. De færreste vil skrolle ned til sidefoten for å finne dette, slik tilfellet er på Helsenorge.
- Språkvalgene må kodes med korrekt språk. Typisk er inngangen til språkvalgene navngitt med synlig eller skjult tekst "Language", som da skal kodes med lang="en", mens de enkelte språkvalgene står på sitt aktuelle språk, type Svenska, Suomi eller Español, og skal kodes med respektive lang-kode. Dette for å sikre korrekt opplesing i hjelpemidler som skjermleser eller opplesende hjelpemidler for personer med dysleksi.

**Comment by Camulos** [comment]
Brreg gikk over til språk-ikon fordi det i ganske mye litteratur på temaet trekkes frem at symbolikk knyttet til geografi ikke nødvendigvis er passende for å indikere språk og språkvalg. Symbolet ble lagt til i Aksel ikon-pakka.
I tillegg har vi (også på bakgrunn av litteraturen) valg å vise knappen for språkvalget med engelsk språk uansett hvilket språk som er valgt.

Vi bruker fremdeles flagg sammen med språkvalgene, selv om det i samme artikler ofte blir poengtert (og det er også ofte testet) at spesielt på språk med mange varianter kan være distanserende og feil når en ser et flagg ved språket sitt. Britisk flagg ble valgt fordi det er britisk engelsk som er det vi lærer på norske skoler og er "offisielt" og er da i det minste noe vi kan begrunne. Uansett ikke optimalt.

Språkvalg står på det språket en viser til, så vi skriver for eksempel ikke "engelsk" når vi viser muligheten til å velge det, men heller "english".
![Skjermbilde 2025-06-06 kl  09 58 48](https://github.com/user-attachments/assets/b0de79d7-1edc-4cc8-865e-665f146d35e2)
![Skjermbilde 2025-06-06 kl  09 59 05](https://github.com/user-attachments/assets/f13e22fd-303b-46cd-b89d-6b8a348970e1)

**Comment by mrosvik** [comment]
Vi har nå brukt språk-ikon en periode på designsystemet.no, men ser at det ikke er alle som oppdager dette. Eksempel: Da jeg sendte lenke til [designsystemet.no](http://designsystemet.no/) til noen i DPGA, så de ikke at det var flere språk og de endte opp med å oversette fra norsk til engelsk via google translate. 

<img width="190" height="115" alt="image" src="https://github.com/user-attachments/assets/2220370a-62f9-483f-8625-a7a269de62b9" />

~~Vi velger derfor å gå tilbake til Globe-ikon, da dette ser ut til å være mest brukt andre steder.~~ Vi tenker også det er viktig at tekst ligger synlig ved siden av ikonet, uansett hvilket ikon som brukes.

**Reply by Camulos** [reply]
Dette gikk ekstremt raskt til en konklusjon uten vurderinger:
Er det symbolet som er problemet?
Er det at det mangler tekst som er problemet?
Er det plasseringen og eller farge av knappen?
Kan det hende at den brukeren som nevnes ikke forventet at det skulle være oversatt og heller ikke så etter en funksjon?

Og så, er det korrekt å gjøre en slik endring på bakgrunn av et enkelt tilfelle?

**Reply by mrosvik** [reply]
Takk @Camulos. Jeg var litt for rask med å skrive her.🙃 Det viktigste er å få inn tekst, så vi lander foreløpig på dette: 
<img width="181" height="80" alt="image" src="https://github.com/user-attachments/assets/da3b5db3-a48b-4f6b-9365-adf87633741a" />

**Argumenter:** 
- Google, og flere, har innarbeidet dette ikonet.
- De andre ikon-alternativene for språk kan ha flere andre betydninger enn språk
- Globe-ikonet handler om lokasjon og geografi, ikonet blir også brukt til helt andre ting.

**Comment by mrosvik** [comment]
## Unngå bruk av flagg
W3C: "Do not use flag icons to indicate languages."

Noen kilder på hvorfor flagg ikke bør brukes på språkvalg:
- https://www.w3.org/TR/i18n-html-tech-lang/#ri20040808.173208643
- https://www.w3.org/International/questions/qa-link-lang#flags

**Comment by Smoerbukk** [comment]
**Erfaringer og refleksjoner fra Udir:**
Etter en vurdert de forskjellige mulighetene, så har vi har valgt å bruke språk-ikonet "文+A"

**Argumenter for**
- Ikonet er lett å skille fra andre ikoner
- Ikonet har ingen andre kjente betydninger
- Andre ikoner vi har vurdert som "globus", "flagg" og andre kan ofte forveksles med andre betydninger enn språk.
- Flagg-ikonet kan bli problematisk, mtp. at ikke alle land kun har ett språk, eller at flere land bruker samme språk

**Argumenter mot**
- Selv om språk-ikonet ikke kan forveksles med andre ikoner, eller har flere kjente betydninger, så er ikonet mindre allment kjent

**Comment by skjalgsm** [comment]
Apple har også retningslinjer på at flagg representer land og ikke språk. For eksempel det å bruke det britiske flagget kan oppleves som ganske kjipt for mange i engelskspråklige land som har en veldig kjip historie med å være en koloni.

I tillegg er det forvirrende når man har flere regionale språk innad i et land (norsk er jo et godt eksempel). Det at det er lettere å gjenkjenne et flagg enn tekst er ikke et godt argument lenger når man har 3 norske flagg i språkmenyen sin.

**Comment by kjartan-neon** [comment]
Når man skal beskrive dette mønsteret, hadde det vært nyttig å også beskrive språkvalg for teksting av video-innhold? Vi har jobbet endel med begge deler i NRK, og bla. kjørt brukertester etc på ordlyd etc.

<img width="920" height="656" alt="image" src="https://github.com/user-attachments/assets/7107ca7a-ba85-49f5-b21f-5de91a096610" />

**Comment by mrosvik** [comment]
[Brukerundersøkelse fra Wikipedia:](https://www.mediawiki.org/wiki/Reading/Web/Desktop_Improvements/Features/Language_switching)
- Det finnes ikke noe universelt, globalt forstått ikon for språkvalg.
- De testet dagens ikon, som kombinerer «A» og «文», mot et globusikon 🌐.
- Av 40 flerspråklige testpersoner forsto 35 at A/文-ikonet handlet om språkvalg, mens bare 13 tolket globusen slik. Resultatet var ikke statistisk signifikant, men tydet klart på at globusen ikke var mer intuitiv.
- Samtidig var A/文-ikonet heller ikke perfekt: omtrent 13 % forsto ikke at det betydde språkvalg.
- Derfor anbefalte de å beholde A/文-ikonet, men bruke en synlig tekstetikett som «Språk» / «Languages» sammen med ikonet når det er mulig.

**Comment by mrosvik** [comment]
## Update from our first pattern meeting

We had our first pattern meeting on language selection today (01.09.26). We looked at how different organisations currently handle language selection and discussed the scope and direction of the pattern.

**Some of the main topics we discussed:**

- Different scenarios: We should distinguish between websites where all content is translated and websites where only parts of the content are available in multiple languages.
- Placement: The recommended placement of the language selector may depend on these different scenarios and the overall structure of the website.
- Language icon: We discussed which icon to recommend for language selection. So far, the 文+A icon is the option for which we have found the strongest supporting research, including Wikimedia's usability testing. We would particularly welcome more research or user insights on this.
- Other topics: URL structure and language codes, mobile behaviour, automatic language selection based on browser/OS preferences, and how to handle content that is not available in the selected language.
- Out of scope: Topics covered by the Norwegian Language Council (Språkrådet) will primarily be referenced rather than covered by this pattern.

We’ll now start drafting the first version of the pattern article and use that as a basis for the next meeting.

If you have research, user insights, examples or experience with language selection, please share them in this thread! We’re especially interested in evidence that can help us validate or challenge the recommendations as they develop.

**Comment by TheTechArch** [comment]
Her er eksempel på hvordan det er implementert med designsystemet. Både for desktop og mobil. Jeg har hele 13 språk og det har vært litt utfordrende å presenter listen over valg på mobil når det er så mange. (har hatt noen forsøk hvor noen språk har havnet utenfor skjermen)
https://evkx.net/evsearch/

Bonus problemstilling er hvordan håndterer man best språk kontra land. 
Dette med når innholdet kan variere på språk og land, men kanskje hvor det er nautrlig å defaulte til land basert på språk. Norsk er f.eks enkelt å default til norge, men engelsk er ikke like sikkert om det er usa, GB, australia, new zealand osv man er ute etter.
